pg_search/src/postgres/storage/metadata.rs

```diff
@@ -191,8 +216,8 @@ impl MetaPage {
     }
 
     pub fn fsm(&self) -> pg_sys::BlockNumber {
-        assert!(block_number_is_valid(self.data.fsm));
-        self.data.fsm
+        assert!(block_number_is_valid(self.data.v2_fsm));
+        self.data.v2_fsm
     }
```

Comment:
In enterprise, on hot standbys, there will be a phase where only `v1` exists, before `v2` exists. Could handle that here to have a slightly smaller patch.